### PR TITLE
feat(Table): implement selection exclude logic

### PR DESCRIPTION
**Pull request by ajkl2533**
Closes FEP-1582

### CODE REVIEW

File: src/components/Datatable/BatchModule/ElementCounter/ElementCounter.tsx [hunk 1]
```diff
@@ -30,22 +44,49 @@ const getCounterContent = (totalLength, selectedLength = 0) => {
...
+  const computedSelectedLength = hasExclusionLogic
+    ? totalLength - selectedLength
+    : selectedLength;
+
+  const content = getCounterContent(totalLength, computedSelectedLength);
+  const handleSelectClick = useCallback(
+    (state) => {
+      setHasExclusionLogic(state);
+    },
+    [setHasExclusionLogic],
```
Comment (PrincAm) [inline src/components/Datatable/BatchModule/ElementCounter/ElementCounter.tsx:64]: For those useCallbacks, is correct assumption that the setter functions never update and will be used just their first instance?

File: src/components/Datatable/Datatable.stories.tsx [hunk 2]
```diff
@@ -181,18 +180,29 @@ export const Default: Story = () => {
     },
     [setIsLoading, setTableData],
   );
+  const logSelectedRows = useCallback((ids, exclude) => {
+    console.log(
+      `OnSelect called for ids: ${ids} to be ${
+        exclude ? 'excluded' : 'included'
+      }`,
+    );
+  }, []);
```
Comment (PrincAm) [inline src/components/Datatable/Datatable.stories.tsx:189]: Here useCallback shouldn't help without dependency

File: src/components/Datatable/BatchModule/ElementCounter/ElementCounter.tsx (hunk 1 above)
Comment (ajkl2533) [reply src/components/Datatable/BatchModule/ElementCounter/ElementCounter.tsx:64]: yep it's internal DT state which isn't customizable and not exposed

File: src/components/Datatable/Datatable.stories.tsx (hunk 2 above)
Comment (ajkl2533) [reply src/components/Datatable/Datatable.stories.tsx:189]: Yep doesn't make much sense here, it doesn't hurt either in this case, will remove

**Review by PrincAm — APPROVED** [review]
LGTM, just a question for better understanding

**Pull request merged**